src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,101 @@
+package nearsoft.academy.bigdata.recommendation;
+
+import org.apache.mahout.cf.taste.impl.model.file.FileDataModel;
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class MovieRecommender {
+
+    private int reviews;
+    private long users;
+    private int products;
+    public UserBasedRecommender recommender;
+    private HashMap<String, Integer> productTable = new HashMap();
+    private HashMap<String, Long> userTable = new HashMap();
+    private HashMap<Integer, String> iProductsTable = new HashMap();
+
+    public MovieRecommender(String txt) throws Exception {
+        readReviewsFile(txt);
+    }
+
+    private void readReviewsFile(String txt) throws Exception {
+        DataModel model;
+        UserSimilarity similarity;
+        UserNeighborhood neighborhood;
+
+        File file = new File(txt);
+        BufferedReader br = new BufferedReader(new FileReader(file));
+        FileWriter writer = new FileWriter("data.csv");
+        BufferedWriter bw = new BufferedWriter(writer);
+        String csv = "";
```

Comment:
Why string? Did you consider a string builder?

Reply:
I didn't consider the inefficiency of string vs. stringBuilder in this case. I'll keep it in mind for the next occasion I do something similar.